src/main/kotlin/christmas/domain/order/Order.kt

```diff
@@ -0,0 +1,30 @@
+package christmas.domain.order
+
+import christmas.domain.menu.Category
+import christmas.domain.menu.Menu
+import christmas.domain.menu.Money
+
+data class Order(val orders: List<Menu>) {
+
+    init {
+        require(orders.size <= 20) { " Order must not greater than 20 orders" }
+        require(orders.filter { order -> order.category == Category.BEVERAGE }.size != orders.size) {
+            "you can not order beverage only"
+        }
+        require(orders.groupBy { it.name }.size == orders.size) {
+            "menu name must not duplicate"
+        }
+    }
+
+    private val _purchaseAmount = orders
+        .map { it.price }
+        .reduce { acc, amount -> acc + amount }
```

Comment:
```suggestion
    private val _purchaseAmount = orders
        .map { it.price }
        .sum()
```

Looks like you made a function for `sum()` but it is not being used.
I didn't test for it but does it work instead?

```kotlin
// Money.kt
fun Iterable<Money>.sum(): Money {
    return this.fold(Money(0L)) { sum, money -> sum + money }
}
```